compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaSymbol.java

```diff
@@ -126,6 +131,23 @@ public Location location() {
         return Optional.ofNullable(this.position);
     }
 
+    @Override
+    public boolean nameEquals(String name) {
+        Optional<String> symbolName = this.getName();
+        if (symbolName.isEmpty() || name == null) {
+            return false;
+        }
+        String symName = symbolName.get();
+        if (name.equals(symName)) {
+            return true;
+        }
+        String unEscapedName = unEscapeIfEscaped(name);
+        if (unEscapedName.equals(symName)) {
+            return true;
+        }
+        return name.equals(unEscapeIfEscaped(symName));
```

Comment:
This doesn't look quite right. For example, consider something like the following:
```ballerina
string 'string;
```
And you try to compare the above var's name with the string `\u{73}tring`. This should return `true`.

Reply:
Oops. My bad. Just corrected. Plese check